Code/Framework/AzCore/AzCore/Component/Component.cpp

```diff
@@ -32,6 +33,7 @@ namespace AZ
     //=========================================================================
     Component::~Component()
     {
+        m_isDeleting = true;
```

Comment:
Would it also work if we stored the `m_entity` in a temp and set it to `nullptr` before going into the if? Something like:
```
auto temp = m_entity;
m_entity = nullptr;
if (temp)
{
    temp->RemoveComponent(this);
}
```
Then we wouldn't need the bool and can keep the the logic for `GetEntityId()` and `GetNamedEntityId()`  the same.

Reply:
That would not completely prevent the crash I am seeing on Linux. If m_entity is set to null, then the original Component::GetEntityId() would continue to the following line:

```
        AZ_Warning("System", false, "Can't get component (type: %s, addr: %p) entity ID as it is not attached to an entity yet!", RTTI_GetTypeName(), this);
        return EntityId();
```
At this point, RTTI_GetTypeName() causes a segmentation fault.